plugins/sinks/stencil/sink_test.go

```diff
@@ -0,0 +1,346 @@
+		err = stencilSink.Sink(ctx, []models.Record{models.NewRecord(data)})
+		assert.Equal(t, errMessage, err.Error())
+	})
+
+	t.Run("should return RetryError if stencil returns certain status code", func(t *testing.T) {
+		for _, code := range []int{500, 501, 502, 503, 504, 505} {
+			t.Run(fmt.Sprintf("%d status code", code), func(t *testing.T) {
+				url := fmt.Sprintf("%s/v1beta1/namespaces/%s/schemas/%s", host, namespaceID, schemaID)
+				client := newMockHTTPClient(map[string]interface{}{}, http.MethodPost, url, stencil.JsonSchema{})
+				client.SetupResponse(code, `{"reason":"internal server error"}`)
+				ctx := context.TODO()
+
+				stencilSink := stencil.New(client, testUtils.Logger)
+				err := stencilSink.Init(ctx, map[string]interface{}{
+					"host":        host,
+					"namespaceId": namespaceID,
+					"schemaId":    schemaID,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				data := &assetsv1beta1.Table{Resource: &commonv1beta1.Resource{}}
+				err = stencilSink.Sink(ctx, []models.Record{models.NewRecord(data)})
+				assert.True(t, errors.Is(err, plugins.RetryError{}))
+			})
+		}
+	})
+
+	successTestCases := []struct {
+		description string
+		data        *assetsv1beta1.Table
+		config      map[string]interface{}
+		expected    stencil.JsonSchema
+	}{
+		{
+			description: "should create the right request to stencil",
+			data: &assetsv1beta1.Table{
+				Resource: &commonv1beta1.Resource{
+					Name:    "stencil",
+					Type:    "table",
+					Service: "bigquery",
+				},
+				Schema: &facetsv1beta1.Columns{
+					Columns: []*facetsv1beta1.Column{
+						{
+							Name:        "id",
+							Description: "It is the ID",
+							DataType:    "INT",
+							IsNullable:  true,
+						},
+						{
+							Name:        "user_id",
+							Description: "It is the user ID",
+							DataType:    "STRING",
+							IsNullable:  false,
+						},
+						{
+							Name:        "description",
+							Description: "It is the description",
+							DataType:    "STRING",
+							IsNullable:  true,
+						},
+					},
+				},
+			},
+			config: map[string]interface{}{
+				"host":        host,
+				"namespaceId": namespaceID,
+				"schemaId":    schemaID,
+			},
+			expected: stencil.JsonSchema{
+				Id:     fmt.Sprintf("%s/%s.%s.json", host, namespaceID, schemaID),
+				Schema: "https://json-schema.org/draft/2020-12/schema",
+				Title:  "stencil",
+				Type:   "table",
+				Properties: map[string]stencil.Property{
+					"id": {
+						Type:        []stencil.JsonType{stencil.JsonTypeNumber, stencil.JsonTypeNull},
+						Description: "It is the ID",
+					},
+					"user_id": {
+						Type:        []stencil.JsonType{stencil.JsonTypeString},
+						Description: "It is the user ID",
+					},
+					"description": {
+						Type:        []stencil.JsonType{stencil.JsonTypeString, stencil.JsonTypeNull},
+						Description: "It is the description",
+					},
+				},
+			},
+		},
+		//{
```

Comment:
@scortier If this piece of code is not needed, let's remove this instead of commenting.

Reply:
Sure, will add more test and refactor remaining part, working on it